subprojects/model-groovy/src/integTest/groovy/org/gradle/cli/TaskOptionsSpec.groovy

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.cli
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class TaskOptionsSpec extends AbstractIntegrationSpec {
+    def "can use a built-in option"() {
+        when:
+        buildScript """
+            abstract class MyTask extends DefaultTask {
+                @Option(option="profile", description="foobar")
+                void setProfile(String value) {
+                    System.out.println("profile=" + value)
+                }
+            }
+
+            tasks.register('mytask', MyTask.class)
+        """
+
+        then:
+        succeeds "mytask", "--profile"
+    }
+
+    def "can use a task option"() {
+        when:
+        buildScript """
+            abstract class MyTask extends DefaultTask {
+                @Option(option="profile", description="foobar")
+                @Optional
+                @Input
+                abstract Property<String> getProfile()
+
+                @TaskAction
+                void run() {
+                    if (getProfile().isPresent()) {
+                        logger.lifecycle("profile=" + getProfile().get())
+                    }
+                }
+            }
+
+            tasks.register('mytask', MyTask.class)
+        """
+
+        then:
+        succeeds "--", "mytask", "--profile", "myvalue"
+        output.contains "profile=myvalue"
+    }
+
+    def "task options apply to most recent task"() {
```

Comment:
I think having 2 different task types here confuses the matter a little.  Maybe change the output to include `getName()` to print the task name, then you can register 2 tasks with the same type and will see which ran as well.

Reply:
No need for "getName()" here, I think -- we can just check that profile=A and profile=B both occur in the output. Happy to use the task name if you feel that it's clearer or better in some way, though.

Reply:
No need, I really just wanted to remove the need for 2 separate task definitions.

Reply:
Ended up adding it here anyway to reduce duplication.